test/jdk/javax/swing/JFileChooser/6798062/bug6798062.java

```diff
@@ -21,73 +21,94 @@
  * questions.
  */
 
-/* @test %W% %E%
-   @bug 6798062
-   @summary Memory Leak on using getFiles of FileSystemView
-   @author Pavel Porvatov
-   @modules java.desktop/sun.awt
-            java.desktop/sun.awt.shell
-   @run applet/manual=done bug6798062.html
-*/
+/*
+ * @test
+ * @bug 6798062
+ * @requires (os.family == "windows")
+ * @summary Memory Leak on using getFiles of FileSystemView
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @modules java.desktop/sun.awt
+ *          java.desktop/sun.awt.shell
+ * @run main/manual bug6798062
+ */
 
 import sun.awt.OSInfo;
 import sun.awt.shell.ShellFolder;
 
-import javax.swing.*;
+import java.awt.Dimension;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
-import java.awt.*;
 import java.io.File;
 import java.io.FileNotFoundException;
-
-public class bug6798062 extends JApplet {
-
-    private final JSlider slider = new JSlider(0, 100);
-
-    private final JTextField tfLink = new JTextField();
-
-    private final JButton btnStart = new JButton("Start");
-
-    private final JButton btnStop = new JButton("Stop");
-
-    private final JButton btnGC = new JButton("Run System.gc()");
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JSlider;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+
+public class bug6798062 {
+
+    private static final String INSTRUCTIONS = """
+            The test is suitable only for Windows.
+
+            1. Create a link
+            2. Copy path to the link into TextField
```

Comment:
Could you amend the instructions to clarify them?

Define what is a _link_.

Reply:
It seems a Windows shortcut is crucial to reproducing the original problem, if it is, the handler for the Start button could show an error message about this fact.

Perhaps, if this link cannot be converted to a file in `MyThread` constructor, `FileNotFoundException` should be propagated and caught in the Start button actionPerformed handler to show an error message and let the tester use another Windows shortcut.

Reply:
SInce this exercise is just to convert applet to main, I tried to keep the instructions same (although I agree it seems ambiguous and I also am not clear what needs to be done) which seems to be ok with the tester since 2009 when the test was written.
I think whatever test change needs to be done should be done in JDK-8146446 as till then this test will not be run..

ELse if you disagree, what you propose to be written in the "instructions"?

Reply:
Create a new bug to improve and to clean up the test then.

Reply:
At the same time, I think you should use the term _“shortcut (.lnk file”_ in the instructions so that it's clear.

The thing is .lnk files are known as shortcuts for most users, even though in Windows Shell API uses link internally which is reflected in the file extension.

Reply:
> Create a new bug to improve and to clean up the test then.

I see, there's an existing bug—[JDK-8146446](https://bugs.openjdk.org/browse/JDK-8146446)—against this test. Although the bug describes that memory usage somewhat grows while the background thread is running, it actually does. It has just grown for me from 100MB to 300MB. Running `System.gc` reduces the memory consumption to 200 MB.